common/src/main/java/org/mvndaemon/mvnd/common/TimeUtils.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mvndaemon.mvnd.common;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.text.NumberFormat;
+import java.time.Duration;
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Time utils.
+ *
+ * Origin file:
+ * https://github.com/apache/camel/blob/4ea9e6c357371682b855d2d79655b41120331b7a/core/camel-util/src/main/java/org/apache/camel/util/TimeUtils.java
+ */
+public final class TimeUtils {
```

Comment:
TimeUtils brings quite a lot of code. I wonder whether I cannot be simplified or perhaps fully replaced by `java.time.Duration.parse(CharSequence)`/`java.time.Duration.toString()`?

Reply:
I'm all for a tiny `mvnd` executable, but I don't think our code size will become a problem at the moment.  When the executable has been trimmed (I'll raise an issue with some infos), we could revisit if needed.

The goal is to be human readable.  The `java.time.Duration.parse`/`toString` methods use the ISO format which looks like `PT8H6M12.345S`, and this is not friendly at all imho.

Reply:
I think the code could be simplified by using a single regex instead of multiple ones.  I'll have a look.

Reply:
I've double checked and the `mvnd` executable already contains the `java.time` packages, mainly because the `java.lang.ProcessHandleImpl$Info.toString()` references it indirectly (in `#totalCpuDuration()`).  I'll try removing this reference to see how much we gain in the final size.

Reply:
> I think the code could be simplified by using a single regex instead of multiple ones.

That would be nice. I even would not mind supporting only the shortest variants `1d` vs. `1 day`, `1 days`.

Reply:
> I've double checked and the `mvnd` executable already contains the `java.time` packages, mainly because the `java.lang.ProcessHandleImpl$Info.toString()` references it indirectly (in `#totalCpuDuration()`). I'll try removing this reference to see how much we gain in the final size.

Interesting excercise :)

Reply:
It's not really easy to get rid of the `java.time.*` packages as those are referenced by various classes in the JDK such as `java.util.Formatter`, `java.util.logging.LogRecord`...